### ISSUE TITLE
Menu bar icon can't be dragged/re-ordered

### ISSUE THREAD

**Issue by erawhctim** [hmarr/vitals#5]
Thanks for creating this neat little app! I've faced a similar problem to what you outlined in your blog post, and I'm excited that I found Vitals.

I'm not sure what causes this behavior, but I can't seem to drag-n-drop the Vitals icon to reorder it among my other menu bar apps. I use a _ton_ of them (iStat Menus, Dropbox, Alfred, BlueJeans, 1Password, to name a few) - all of them can be reordered if I hold down CMD and click to drag. Vitals seems to treat this as a regular click (and opens the main processes window. 

I may poke around to see if I can figure it out, but I have no experience with menu bar apps so I don't expect to find a solution quickly 😅

**Comment by hmarr** [comment]
Nice catch! I suspect (hope) it'll just be a case of checking whether cmd is being held down in the [click event handler](https://github.com/hmarr/vitals/blob/2dad6fe060ff1fe6342edb59d297b1542694e5ca/Vitals/StatusBarController.swift#L42-L58). 

At some point I'd like to try embedding the UI in a regular `NSMenuItem` (by setting its `view` property) rather than using an `NSPopover`. I think that'd mean all the native behaviour works as expected out of the box, and it'd also mean I could put the "Quit" menu item there too – right now you need to right click the menu bar item to see it which isn't super intuitive.

**Comment by erawhctim** [comment]
That makes sense and seems to align with what I've read from some cursory googling. 

on another note - how do I build/run this locally? I pulled it up in Xcode and got an error that led me to download v12 beta, but that doesn't seem to run on my version of macOS (10.15.7). Am I SOL until I can run Catalina, or is there a workaround?

**Comment by hmarr** [comment]
Hmm, what's the error you're getting? I'm using Xcode 12.3 with Catalina 10.15.7.

**Comment by erawhctim** [comment]
Oh! 🤦 I was using the 12.5 beta, that explains it. 

I downloaded 12.4 and can run it now 🙂

**Comment by hmarr** [comment]
Fortunately the [fix](https://github.com/hmarr/vitals/commit/845a31de63b9dde61351db1a4abb7d68c119805d) was pretty simple! Thanks again for pointing this out. I'll cut a new release soon.

**Comment by erawhctim** [comment]
Oh nice! Love to see that.

On March 23, 2021, BasheerMalaaNBCU ***@***.***> wrote:
> Fortunately the fix
> <https://github.com/hmarr/vitals/commit/845a31de63b9dde61351db1a4abb7d68c119805d>
> was pretty simple! Thanks again for pointing this out. I'll cut a new
> release soon.
>

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/hmarr/vitals/issues/5#issuecomment-805378846>, or
unsubscribe <https://github.com/notifications/unsubscribe-
auth/ABSLRVYG7HD4PXVXA4WP2VTTFEXL3ANCNFSM4ZT36ZFQ>.

**Comment by hmarr** [comment]
I just published v0.2, which includes this fix 🚀 

If you're using homebrew:

```
brew tap hmarr/tap  # if you haven't already done this
brew update         # if you're updating from v0.1
brew install vitals
```

Otherwise you can download v0.2 from the [releases page](https://github.com/hmarr/vitals/releases).
